File: ABC_400/ABC422/B.py
H, W = map(int,input().split())
S = [input() for _ in range(H)]

directions = [(-1, 0), (1, 0), (0, -1), (0, 1)]

for i in range(H):
    for j in range(W):
        if S[i][j] == "#":
            cnt = 0
            for di, dj in directions:
                ni, nj = i + di, j + dj
                if 0 <= ni < H and 0 <= nj < W and S[ni][nj] == "#":
                    cnt += 1
            if cnt not in [2, 4]:
                print("No")
                exit()

print("Yes")

File: ABC_400/ABC422/B.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int H, W;
    cin >> H >> W;
    vector<string> S(H);
    for (int i = 0; i < H; i++) cin >> S[i];

    // 上下左右の方向ベクトル
    int dx[] = {-1, 1, 0, 0};
    int dy[] = {0, 0, -1, 1};

    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            if (S[i][j] == '#') {
                int cnt = 0;
                for (int d = 0; d < 4; d++) {
                    int ni = i + dx[d];
                    int nj = j + dy[d];
                    if (0 <= ni && ni < H && 0 <= nj && nj < W && S[ni][nj] == '#') {
                        cnt++;
                    }
                }
                // 2または4でなければ不適合
                if (cnt != 2 && cnt != 4) {
                    cout << "No" << endl;
                    return 0;
                }
            }
        }
    }

    cout << "Yes" << endl;
    return 0;
}
